include/api/CAnomalyJobConfig.h

```diff
@@ -90,8 +92,13 @@ class API_EXPORT CAnomalyJobConfig {
         using TDetectorConfigVec = std::vector<CDetectorConfig>;
 
     public:
+        //! Default constructor
         CAnalysisConfig() {}
 
+        //! Constructor taking a map of detector rule filters keyed by filter_id.
+        explicit CAnalysisConfig(const CFieldConfig::TStrPatternSetUMap& ruleFilters)
```

Comment:
It would be better to make this `CDetectionRulesJsonParser::TStrPatternSetUMap` - it's exactly the same type but avoids mentioning the legacy class `CFieldConfig` in this file.